#72 Implement  Two factor authentication for Vendor and Admin

Shumba250 opened the pull request:
#### What does this PR do?

- Implement a two-factor authentication page for vendor login
#### Description of Task to be completed

- Implement a responsive two-factor authentication page
- Process OTP and redirect the vendor to the dashboard
- Validate the OTP input field
#### How should this be manually tested?

- clone the repo using git clone repo's URL
- change branch git checkout ft-two-factor-auth
- create .env file and paste variables from .env.example
- run npm install 
- run npm run dev
- signup a vendor and login
- Copy the OTP sent to your email and paste it in the input filed

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

Shumba250 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Seller auth & Two Factor Authentication for Sellers](https://trello.com/c/K58Vpe6U/72-seller-auth-two-factor-authentication-for-sellers)

oliviertech212 committed:
feat(signin):signin with email and password
-add style
[WIP]

sostene12 reviewed (commented):
<img width="1280" alt="Screenshot 2023-05-01 at 14 49 06" src="https://user-images.githubusercontent.com/88575509/235453860-1c7edc01-b6b7-4912-9531-468ea723d23e.png">
this card is soo big, if you can reduce it

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

sostene12 commented on coverage/components/Auth/Input.jsx.html:
i suggest you don't push coverage folder, please ignore it

sostene12 commented on src/pages/index.jsx:
consider using app.jsx as the main file of routes

sostene12 commented on src/redux/actions/authAction.js:
it's not a good practice to use browser storage mean, if it is local storage stick with it only, or stick to using cookies only

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success

Shumba250 committed:
ft(two-factor-auth for sellers)
implement a two factor authentication page for vendors
implement OTP input field validation
implement OTP validation and redirect the seller to dashboard on success
